components/object_store/src/obkv/mod.rs

```diff
@@ -0,0 +1,940 @@
+        let scan_request = util::scan_request_with_prefix(prefix.as_bytes());
+
+        let mut iter = self
+            .client
+            .scan(scan_context, table_name, scan_request)
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+
+        let mut keys = vec![];
+        while iter.valid() {
+            keys.push(iter.key().to_vec());
+            iter.next().map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+        }
+
+        self.client
+            .delete_batch(table_name, keys)
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+
+        // Here to delete meta with path `location` and multipart_id
+        self.meta_manager
+            .delete_with_version(location, multipart_id)
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+
+        debug!(
+            "ObkvObjectStore abort_multipart operation, location:{location}, table_name:{table_name}, cost:{}",
+            instant.elapsed().as_millis()
+        );
+        Ok(())
+    }
+
+    async fn get(&self, location: &Path) -> Result<GetResult> {
+        let instant = Instant::now();
+        let result = self.get_internal(location).await;
+
+        debug!(
+            "ObkvObjectStore get operation, location:{location}, cost:{}",
+            instant.elapsed().as_millis()
+        );
+        result.box_err().map_err(|source| StoreError::NotFound {
+            path: location.to_string(),
+            source,
+        })
+    }
+
+    async fn get_range(&self, location: &Path, range: Range<usize>) -> Result<Bytes> {
+        let instant = Instant::now();
+
+        let table_name = self.pick_shard_table(location);
+        let meta =
+            self.read_meta(location)
+                .await
+                .box_err()
+                .map_err(|source| StoreError::NotFound {
+                    path: location.to_string(),
+                    source,
+                })?;
+
+        let mut range_buffer = Vec::with_capacity(range.end - range.start);
+        let covered_parts = meta
+            .compute_covered_parts(range)
+            .box_err()
+            .map_err(|source| StoreError::NotFound {
+                path: location.to_string(),
+                source,
+            })?;
+
+        for (index, key) in covered_parts.part_keys.iter().enumerate() {
+            let values = self
+                .client
+                .get(table_name, key.as_bytes())
+                .map_err(|source| StoreError::NotFound {
+                    path: location.to_string(),
+                    source: Box::new(source),
+                })?;
+
+            if let Some(bytes) = values {
```

Comment:
Shall throw an error if `values` is None?

Reply:
Done